tf2onnx/onnx_opset/tensor.py

```diff
@@ -1099,11 +1105,19 @@ def version_1(cls, ctx, node, **kwargs):
 
         attr = {"axes": slice_axis, "ends": ends, "starts": starts}
         inputs_map = {"data": trans2.output[0], **attr}
-        kwargs = {**inputs_map, "outputs": node.output}
         dtypes = [ctx.get_dtype(node.output[0])]
-        shapes = [ctx.get_shape(node.output[0])]
-        ctx.remove_node(node.name)
-        GraphBuilder(ctx).make_slice(kwargs, name=node.name, dtypes=dtypes, shapes=shapes)
+        shapes = ctx.get_shape(node.output[0])
+
+        if len(ctx.get_shape(input_tensor.output[0])) == 3:
+            # add a squeeze op to convert output into 3d
+            kwargs = {**inputs_map}
+            ctx.remove_node(node.name)
```

Comment:
put duplicate `ctx.remove_node` outside if scope.

Reply:
I can't put duplicate ctx.remove_node outside if scope, because make_slice kwargs for 4d input is 
`kwargs = {**inputs_map, "outputs": node.output} `
and make_slice kwargs for 3d input is 
`kwargs = {**inputs_map}`
I need to `ctx.remove_node` after setting make_slice kwargs